src/main/recorder.ts

```diff
@@ -280,6 +289,52 @@ type RecorderOptionsType = {
         obsRecorder.reconfigure(this._options);
         if (mainWindow) mainWindow.webContents.send('refreshState');
     }
+
+    /**
+     * Generate a filename for the final video of a recording according
+     * to its category, if possible.
+     *
+     * The final filename should contain adequate information to quickly be able
+     * to identify the video files on the filesystem and be able to tell what it
+     * was about and the result of it.
+     */
+    getFinalVideoFilename (metadata: Metadata): string | undefined {
+        let outputFilename = '';
+        let zoneName: string;
+        const resultText = getVideoResultText(metadata.category, metadata.result);
+
+        switch (metadata.category) {
+            case VideoCategory.TwoVTwo:
+            case VideoCategory.ThreeVThree:
+            case VideoCategory.Skirmish:
+            case VideoCategory.SoloShuffle:
+                zoneName = getInstanceNameByZoneId(metadata.zoneID);
+                outputFilename = `${zoneName} (${resultText})`;
+            break;
+
+            case VideoCategory.Raids:
+                const encounterName = getEncounterNameById(metadata.encounterID);
+                zoneName = getRaidNameByEncounterId(metadata.encounterID);
+                outputFilename = `${zoneName}, ${encounterName} (${resultText})`;
+            break;
+
+            case VideoCategory.MythicPlus:
+                if (metadata.challengeMode) {
+                    const cm = metadata.challengeMode;
+                    const keystoneUpgradeLevel = ChallengeModeDungeon.calculateKeystoneUpgradeLevel(cm.allottedTime, cm.duration);;
```

Comment:
This case could probably do with its own helper function `getChallengeModeResultText` or so, there is a bit much packed in here for my liking. VideoButton could also benefit from sharing it?

Reply:
VideoButton wouldn't benefit from this, as it's only used for the filename, really. It's not so much result text (which we have already "_Depl_" and "_Time_") as it is a condensed string of information for the given video.

I can surely give it its own method!